prog/dftb+/lib_dftb/etemp.F90

```diff
@@ -464,7 +514,9 @@ subroutine electronFill(Eband, filling, TS, E0, Ef, eigenvals, kT, distrib, kWei
 #:else
             filling(j, i, iSpin) = 1.0_dp / (1.0_dp + exp(x))
 #:endif
-            if (filling(j, i, iSpin) <= elecTol) then
+            if (tNonAufbau .and. j/=1 .and. ((filling(max(j-1,1), i, iSpin)+filling(j-1, i, iSpin))) <= elecTol) then
```

Comment:
This line might still have a bug, as the loop runs j=1,... and filling(j-1, ) is referenced by some compilers even if the j/=1 test fails. Also, most of the time max(j-1,1) = j-i so the same level is tested twice.
I think the line  should probably read
if (tNonAufbau .and. j/=1 .and. ((filling(max(j-1,1), i, iSpin)+filling(j, i, iSpin))) <= elecTol) then